slim-cli.py

```diff
@@ -197,14 +214,59 @@ def read_file_content(file_path: str) -> Optional[str]:
         logging.error(f"Error reading file {file_path}: {e}")
         return None
 
+import os
+import requests
+from azure.identity import ClientSecretCredential
+from dotenv import load_dotenv
+
+load_dotenv()
+
+TENANT_ID = os.getenv("AZURE_TENANT_ID")
+CLIENT_ID = os.getenv("AZURE_CLIENT_ID")
+CLIENT_SECRET = os.getenv("AZURE_CLIENT_SECRET")
+API_ENDPOINT = os.getenv("API_ENDPOINT")
+DEPLOYMENT_ID = os.getenv("DEPLOYMENT_ID")
+
+authority = "https://login.microsoftonline.com"
+credential = ClientSecretCredential(
```

Comment:
Here - you're initializing Azure services even though no arguments to slim-cli have been passed yet. You need to only initialize Azure services if someone picks that model.

e.g. I'm currently getting this error without any arguments:
```
% python slim-cli.py                            
Traceback (most recent call last):
  File "slim-cli.py", line 231, in <module>
    credential = ClientSecretCredential(
  File "/venv/lib/python3.9/site-packages/azure/identity/_credentials/client_secret.py", line 45, in __init__
    raise ValueError("client_id should be the id of a Microsoft Entra application")
ValueError: client_id should be the id of a Microsoft Entra application
```